autoupdate.js

```diff
@@ -0,0 +1,57 @@
+const autoUpdater = require('electron-updater').autoUpdater
+const { dialog } = require('electron');
+
+const hour = 60 * 60;
+const autoUpdaterInterval = hour * 1000;
+
+const RESTART_BUTTON = 0;
+const LATER_BUTTON = 1;
+
+function autoUpdateDisabled(config) {
+  return process.mas || config.get('disableAutoUpdate');
+}
+
+function checkForUpdates() {
+  autoUpdater.checkForUpdates();
+}
+
+function showUpdateDialog() {
+  const options = {
+    type: 'info',
+    buttons: ['Restart', 'Later'],
+    title: "Signal update available",
+    message: "There is a new version of Signal available.",
+    detail: "Press Restart to apply the updates.",
+    defaultId: RESTART_BUTTON,
+    cancelId: LATER_BUTTON
+  }
+
+  dialog.showMessageBox(options, function(response) {
+    if (response == RESTART_BUTTON) {
+      autoUpdater.quitAndInstall();
+    }
+  });
+}
+
+function onUpdateAvailable() {
+  showUpdateDialog();
+}
+
+function onError(error) {
+  console.log("Got an error while updating: ", error);
+}
+
+function initializeAutoUpdater(config) {
+  if (autoUpdateDisabled(config)) { return; }
+
+  autoUpdater.addListener('update-available', onUpdateAvailable);
```

Comment:
Yeah, you're totally right. Let me update this and push a commit.

Side note: Would you be able to give me a hand testing this on Linux and Windows if you have test machines handy? Otherwise I'll spend some time getting VMs up and running. Thanks!